app/controllers/types_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TypesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_type, only: [:show, :edit, :update, :destroy]
+  respond_to :js, :html
+
+  def index
+    @types = current_user.types.order('created_at DESC').page(params[:page]).per(4)
+
+    @types = if params[:show_all] == 'true'
+               Type.all.order('created_at DESC').page(params[:page]).per(4)
+             else
+               current_user.types.order('created_at DESC').page(params[:page]).per(4)
+             end
+  end
+
+  def show
+  end
+
+  def new
+    @type = Type.new
+    @type.user_id = current_user.id
+  end
+
+  def edit
+  end
+
+  def create
+    @type = Type.new(type_params)
+    @type.user_id = current_user.id
```

Comment:
@VanjaJeftic usually it's better to perform `build`/`new` through an association. It also sets the user_id and it has the added benefit that the record is immediately available on `current_user` without having to reload that object.

```suggestion
    @type = current_user.types.new(type_params)
```